From 53e44fc5bab34e4846b5276795403ef36f61d48b Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Wed, 12 Mar 2025 15:59:27 +0100
Subject: [PATCH 819/954] feat(st-pmic): add default regulator configuration
 for enable_ramp_delay

Configure the default ramp delay for the STPMIC25, STPMIC2L and STPMIC1L
regulators, the DT configuration should be used only to change this
values, for example for board or external device constraints.

The behavior is aligned with STPMIC1 driver and the 1ms value is aligned with OP-TEE driver.
This patch allows to align the regulator part of the device tree
between OP-TEE and TF-A.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 94aed05fa395024d294aea5cbfe093eebedf8389 feat(st-pmic): add default regulator configuration for enable_ramp_delay]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I97f00fe118f89d45b3d7bc80ca11aa2e74d9e06c
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/443763
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/st/pmic/stm32mp_pmic2.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/st/pmic/stm32mp_pmic2.c b/drivers/st/pmic/stm32mp_pmic2.c
index 247b342ae7..827472d940 100644
--- a/drivers/st/pmic/stm32mp_pmic2.c
+++ b/drivers/st/pmic/stm32mp_pmic2.c
@@ -409,6 +409,7 @@ static struct regul_handle_s pmic2_regul_handles[STPMIC2_NB_REG] = {
 	.node_name = name, \
 	.ops = &pmic2_ops, \
 	.driver_data = &pmic2_regul_handles[rid], \
+	.enable_ramp_delay = 1000, \
 }
 
 #define DEFINE_GPO(rid, name) \
-- 
2.53.0

